client/src/modules/events/pages/eventsPage.tsx

```diff
@@ -51,41 +51,49 @@ export default function Pagination({
     </Flex>
   );
 }
-
+const pageSize = 5;
 export const EventsPage: NextPage = () => {
-  const { loading, error, data } = useMinEventsQuery();
   const [currentPage, setCurrentPage] = useState(1);
-  const pageSize = 5;
-  const paginated = useMemo(() => {
-    const offset = (currentPage - 1) * pageSize;
-    return data?.events.slice(offset, offset + pageSize); // 0 , 4 // 5, 10
-  }, [currentPage, pageSize, data?.events]);
+  const { loading, error, data, fetchMore } = useHomeQuery({
+    variables: { offset: (currentPage - 1) * pageSize, limit: pageSize },
+  });
```

Comment:
`useHomeQuery` works, but it's doing too much: it's getting the chapters, too.  Instead, it would be good to use a dedicated hook so that we only request what we need. Namely, the events.

Since we already have the `useMinEventsQuery` hook, we can modify that to return paginated events and we're good.

The process is a little involved, but what you can do is start with

https://github.com/freeCodeCamp/chapter/blob/84c2f6152d36d4a6dcc7ada94a838b4161073a1a/client/src/modules/events/graphql/queries.ts#L3-L4

(which should be called `eventFeed`, rather than `minEvents`) and modify it to use the `paginatedEvents` resolver.  Once that's changed, `graphql.tsx` should regenerate (if not, `npm run gen` will do that).  That will create a new hook you can use in `eventsPage` to get the paginated data.

Reply:
created  usePaginatedEventsQuery,
thing that currently bother me - i still need to use `minEventsQuery` for getting total events number

Reply:
Yeah, I see the problem.  

What you can do is use `prisma.events.count()` in the `paginatedEvents` resolver to get the total and then return an ObjectType including the events and their count.  There's a few ways to do this, depending on the shape of the data you want on the client.  A single object containing an array of events and a single count seems reasonable to me, but whatever works.